Src/FluentAssertions/Primitives/HttpResponseMessageAssertions.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System.Diagnostics;
+using System.Net;
+using System.Net.Http;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Primitives
+{
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="HttpResponseMessage"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions
+        : HttpResponseMessageAssertions<HttpResponseMessageAssertions>
+    {
+        public HttpResponseMessageAssertions(HttpResponseMessage value)
+            : base(value)
+        {
+        }
+    }
+
+    /// <summary>
+    /// Contains a number of methods to assert that a <see cref="HttpResponseMessage"/> is in the expected state.
+    /// </summary>
+    [DebuggerNonUserCode]
+    public class HttpResponseMessageAssertions<TAssertions>
+        where TAssertions : HttpResponseMessageAssertions<TAssertions>
+    {
+        public HttpResponseMessageAssertions(HttpResponseMessage value) => Subject = value;
+
+        /// <summary>
+        /// Gets the object which value is being asserted.
+        /// </summary>
+        public HttpResponseMessage Subject { get; }
+
+        /// <summary>
+        /// Asserts that the <see cref="HttpStatusCode"/> is successful (2xx).
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        public AndConstraint<TAssertions> BeSuccessful(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
```

Comment:
❌ If the call to `BeSuccessful` was encapsulated by an `AssertionScope`, the `FailWith` will just queue an error message on the current scope, but not throw an actual exception. You either need to chain the two assertions using `Then` or account for the boolean return value of `FailWith`. Same for the other tests.